libraries/shared/storage/fetcher/geth_rpc_storage_fetcher_test.go

```diff
@@ -155,40 +164,172 @@ var _ = Describe("Geth RPC Storage Fetcher", func() {
+				close(done)
+			})
+
+			It("adds errors to error channel if decoding the state diff RLP fails", func(done Done) {
+				streamer.SetPayloads(badStateDiffPayloads)
+
+				go statediffFetcher.FetchStorageDiffs(storagediffChan, errorChan)
+
+				Expect(<-errorChan).To(MatchError("EOF"))
+
+				close(done)
+			})
+
+			It("adds parsed statediff payloads to the out channel", func(done Done) {
+				statediffFetcher = fetcher.NewGethRpcStorageFetcher(streamer, statediffPayloadChan, fetcher.NewGethPatchWithService)
+				streamer.SetPayloads([]filters.Payload{old_geth_patches.MockStatediffPayload})
+
+				go statediffFetcher.FetchStorageDiffs(storagediffChan, errorChan)
+
+				height := test_data.BlockNumber
+				intHeight := int(height.Int64())
+				expectedDiff1 := types.RawDiff{
+					HashedAddress: crypto.Keccak256Hash(test_data.ContractLeafKey[:]),
+					BlockHash:     common.HexToHash("0xfa40fbe2d98d98b3363a778d52f2bcd29d6790b9b3f3cab2b167fd12d3550f73"),
+					BlockHeight:   intHeight,
+					StorageKey:    crypto.Keccak256Hash(test_data.StorageKey),
+					StorageValue:  common.BytesToHash(test_data.SmallStorageValue),
 				}
-				streamer.SetPayloads([]statediff.Payload{badStatediffPayload})
+				expectedDiff2 := types.RawDiff{
+					HashedAddress: crypto.Keccak256Hash(test_data.AnotherContractLeafKey[:]),
+					BlockHash:     common.HexToHash("0xfa40fbe2d98d98b3363a778d52f2bcd29d6790b9b3f3cab2b167fd12d3550f73"),
+					BlockHeight:   intHeight,
+					StorageKey:    crypto.Keccak256Hash(test_data.StorageKey),
+					StorageValue:  common.BytesToHash(test_data.LargeStorageValue),
+				}
+				expectedDiff3 := types.RawDiff{
+					HashedAddress: crypto.Keccak256Hash(test_data.AnotherContractLeafKey[:]),
+					BlockHash:     common.HexToHash("0xfa40fbe2d98d98b3363a778d52f2bcd29d6790b9b3f3cab2b167fd12d3550f73"),
+					BlockHeight:   intHeight,
+					StorageKey:    crypto.Keccak256Hash(test_data.StorageKey),
+					StorageValue:  common.BytesToHash(test_data.SmallStorageValue),
+				}
+
+				diff1 := <-storagediffChan
+				diff2 := <-storagediffChan
+				diff3 := <-storagediffChan
+
+				Expect(diff1).To(Equal(expectedDiff1))
+				Expect(diff2).To(Equal(expectedDiff2))
+				Expect(diff3).To(Equal(expectedDiff3))
+
+				close(done)
+			})
+
+			It("adds errors to error channel if formatting the diff as a StateDiff object fails", func(done Done) {
+				stateDiff := old_geth_patches.StateDiffWithBadStorageValue
+				stateDiffRlp, err := rlp.EncodeToBytes(stateDiff)
+				Expect(err).NotTo(HaveOccurred())
+				payloadToReturn := filters.Payload{StateDiffRlp: stateDiffRlp}
+
+				streamer.SetPayloads([]filters.Payload{payloadToReturn})
 
 				go statediffFetcher.FetchStorageDiffs(storagediffChan, errorChan)
 
-				Expect(<-errorChan).To(MatchError("rlp: input contains more than one value"))
+				errFromChan := <-errorChan
+				Expect(errFromChan.Error()).To(MatchRegexp("rlp: input contains more than one value"))
```

Comment:
super minor but maybe this could be:
```
Expect(errFromChan).To(MatchError(rlp.ErrMoreThanOneValue))
```
to avoid failing if they reword upstream